src/app/test-results/test-results-list/test-results-detail-list.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ResponseTestResultDetail, TestResultService } from 'api-client';
+
+@Component({
+  selector: 'wh-test-results-detail-list',
+  templateUrl: './test-results-detail-list.component.html',
+})
+export class TestResultsDetailListComponent implements OnInit {
+  @Input() buildId: number;
+  @Input() summaryId: number;
+  details: ResponseTestResultDetail[];
+  failedDetails: ResponseTestResultDetail[];
+  private hasFetched: boolean;
+
+  constructor(
+    private testResultService: TestResultService) { }
+
+  ngOnInit(): void {
+    this.fetchTestResultDetails();
+  }
+
+  private fetchTestResultDetails() {
+    if (this.hasFetched) {
+      return;
+    }
+
+    this.hasFetched = true;
+    this.testResultService
+      .getBuildTestResultDetailList(this.buildId, this.summaryId)
+      .subscribe(details => {
+        this.details = details.list;
+        this.details.forEach(d => d.message = this.unescapeHtml(d.message));
+        this.failedDetails = this.details.filter(d => d.status === ResponseTestResultDetail.StatusEnum.Failed);
+      });
+  }
+
+  private unescapeHtml(str: string): string {
+    return str?.replace(/&lt;/g, '<').replace(/&gt;/g, '>').replace(/&amp;/g, '&');
+  }
```

Comment:
As you've now created an issue on this, I think you should remove this. 

Alternatively if you surely insist, then add a comment making it clear that this should be removed later. But having code in the frontend to fix mistakes in the backend is a hard no from me.

Reply:
Removed it. 👍🏻